Talkpod A36Plus support

Jamie793 opened the pull request:
So far. I have implemented BK4819 and BK1080. It can TX and RX. The BK4819 function includes "CDCSS, CTCSS, Scan Frequency, VOX and so on". The BK1080 function includes  "Tune".

silseva committed:
CMSIS library files for Artery AT32F421 MCU

silseva committed:
Radtel RT-890 build target

silseva committed:
Boot code for AT32F421

silseva committed:
Gpio driver for AT32F421 MCU

silseva committed:
Fix compilation warning generated by __set_FPSCR

When compiling for ARM Cortex M4 without FPU, the input parameter of the
intrinsic function __set_FPSCR generates an unused variable warning. To
avoid this, the parameter is now casted to void when FPU is not present or
not used.

silseva committed:
Platform BSP for Radtel RT-890

silseva committed:
Delays driver for AT32F421 MCU

silseva committed:
AT32F421: added linker script section to pad the binary image to an integer multiple of 128 bytes

silseva committed:
USART1 driver for AT32F421 MCU

silseva committed:
AT32F421: initializing USART1 at early boot

silseva committed:
AT32F421: enabled stdout over USART1

silseva committed:
Display: added API functions for color fill and single-pixel coloring

silseva committed:
Graphics: added support for direct on-screen draw without framebuffer

Added support for direct on-screen draw of graphic features. This feature is
controlled by the macro CONFIG_GFX_NOFRAMEBUF. When defined, no framebuffer is
instantiated and all the graphics functions draw directly on the screen.

silseva committed:
Display driver for Radtel RT-890

silseva committed:
Keyboard driver for Radtel RT-890

silseva committed:
FIXME/WIP exclude VP and codec2 to allow compilation

Commit:
a36plus: Initial work

Commit:
Fix stray #endif

Commit:
Further work

Commit:
Fix CMSIS headers

Commit:
GPIO driver fix

Commit:
Fix GPIO pinmap to not brick your device

Commit:
Framebuffer: return to normal

Commit:
graphics: Flip colors back to RGB

Commit:
Fix weird display offset

Commit:
Screen resolution fix

Commit:
Keyboard driver now works

Commit:
Account for bootloader

Commit:
a36plus: Add SPI from GD32F3x0 CMSIS

Commit:
a36plus: SPI init work

Commit:
a36plus: Speed up gfx_fillScreen

Commit:
UI: get rid of some clearscreens for now

Commit:
More UI clearscreen work

Commit:
a36plus: Use SPI to render display

Commit:
Change ld script and BSP to account for bootloader

Commit:
a36plus: Swap columns and rows; add macro key

Commit:
a36plus: GPIO driver update (gpio-native)

Commit:
a36plus: Delay fix

Commit:
a36plus: fix "GPIO driver update"

Commit:
Fix color_t conversion

Jamie793 committed:
added baseband for A36Plus and removed Stub

Jamie793 committed:
porting bk4819

Commit:
Hacks to make the display rendering more bearable for testing

Commit:
Fixed BK4819 SCK pin

Commit:
BK4819, initial working RSSI

Commit:
Implement windowing for partial rendering

Commit:
a36plus: Add BK4819 driver to meson.build

Jamie793 committed:
added bk4819 init and implemented rx&tx

Jamie793 committed:
added bk4819.c

Commit:
Add mic to pinmap

Commit:
Try to fix macro key

Commit:
Make frequency input actually readable

Jamie793 committed:
implemented Tx CTCSS

Jamie793 committed:
fixed the pinmap

Commit:
radio_a36plus: Minor tweaks

Jamie793 committed:
refactored some function and implemented setctc2

Jamie793 committed:
implemented get ctcss and added enable function signature

Jamie793 committed:
implemented Rx CTCSS

Jamie793 committed:
implemented VOX

Jamie793 committed:
renamed vox

Jamie793 committed:
added vox enabled

Jamie793 committed:
implemented Tx CDCSS

Jamie793 committed:
removed unsed function and moved comment

Jamie793 committed:
implemented BK1080

Jamie793 committed:
implemented scanning frequency

Jamie793 committed:
Refactored something

Jamie793 committed:
implementing backlight adjusting

Jamie793 committed:
implemented brightness adjusting

Jamie793 committed:
modified timer compare register value

froqstar commented:
Can/will this target eventually support M17, and if so will it require a hardware mod?

Jamie793 committed:
enabled brightness adjusting

Jamie793 committed:
added openocd config file

Tunas1337 commented:
> Can/will this target eventually support M17, and if so will it require a hardware mod?

It's only an analog radio, so it would be an interesting feat to make M17 work on it. The answer is "maybe." I'll try to get Codec2 working on it for the voice prompts first of all, and then check whether the MCU already has a mic path into the BK4819. If not, yes, it would require a hardware mod.

Tunas1337 committed:
Clear screen after splash screen

Jamie793 committed:
Implemented Rx Tx OFF

Jamie793 committed:
Implemented setting frequency and init, test ok

Jamie793 committed:
Testing BK1080

Jamie793 committed:
nothing

Jamie793 committed:
Implemeted seeking channel

Commit:
Restore flash base to account for bootloader

Commit:
bk1080: Add set frequency function

Commit:
Revert "Testing BK1080"

This reverts commit fd6a268deafd6712cbbf804caca273986b8951e5.

Commit:
Fix About screen

Commit:
Fix compilation error

Commit:
BK4819, various fixes related to RX

Commit:
a36plus: Keyboard driver fully working now

Commit:
a36plus: Power button functionality, beeps

Commit:
a36plus: Initial work on filter/amp switching

Commit:
a36plus: Refactor bk1080 driver

Commit:
a36plus: Screen drawing optimizations

Commit:
a36plus: Change frequency range for testing, add mic level

Commit:
Misc fixes, and proper power amplifier/filter implementation (no UHF TX yet)

Commit:
Show mic level instead of (for now useless) squelch

Commit:
Macro key is now side 1, avoids conflict with number 7

Commit:
Add more power levels: 1W, 5W, 10W

Commit:
No need to pad the image

Commit:
Fix clear_window

Commit:
Working squelch; OpenRTX squelch now tied to BK4819 squelch

Commit:
Improve responsiveness by microseconds

Commit:
Moar power! (improves responsiveness of UI)

Commit:
Add lighter printf library, small fixes

Commit:
Fix power-off busy wait

Commit:
Flash driver and persistence of settings/channels

Commit:
ADC/battery voltage driver

Commit:
UI changes, add callsign, bugfixes

Commit:
Pinmap fixes

Commit:
Remove debug print string

Commit:
Responsiveness tweak in Info screen

Commit:
Nuked idle drain (13mA -> 13µA)

Commit:
Added name to authors list

Commit:
Info screen fix

Commit:
Fix battery upper bound voltage

Commit:
Changes to make other platforms not glitch out

Commit:
Fix UHF TX

Commit:
Minor non-functionality tweaks

Commit:
Initial spectrum analyzer (experimental)

Commit:
Reduce speaker gain to make the volume knob less abrupt

Commit:
Fix VHF RX

Commit:
More forgotten spectrum things

Commit:
NVM driver, remove cruft

Commit:
Update meson.build

Commit:
Waterfall spectrum implementation; should fix compiling

Commit:
Not yet ;) - remove WFM OpMode from meson.build

Commit:
Fix spectrum up/down functionality

Commit:
Reduce overclock, as for some reason it helps with spectrum performance if I do

Commit:
Fix power according to Lukasz, improve RSSI get

Commit:
Spectrum fixes

Commit:
Macro mode fixes

Commit:
Added .kdhx to gitignore as it is just a binary

Commit:
Fix standby on spectrum, plus some other improvements

Commit:
Fixed keyboard driver (again!)

Commit:
Miscellaneous fixes

Commit:
BK4819 stability improvements

Commit:
Channel storage implementation

Commit:
Miscellaneous fixes, add menu to spectrum

Commit:
Fix Cortex M4 compilation

Commit:
Rewrote code that went missing, and resubmitting it

Commit:
Fix missing braces on if statement

Commit:
a36plus: AM/FM receive mode switch

Commit:
Frequency no longer flickers on state update

Commit:
Disable level meter marks (they don't show up)

Commit:
EXPERIMENTAL: Use calibration data for TX power (PA Bias)

Commit:
ui: Fix bottom part not being rendered when macro menu is open in channel mode

Commit:
a36plus/keyboard: KEY_MONI can now be used alongside other keys

Commit:
Minor fixes, improve spectrum scanning speed by more than 2x

Commit:
a36plus: add flashing and wrapping to meson.build

Commit:
Remove beep (not yet)

Commit:
Fix AM mode display in FM mode and macro menu

Commit:
Expand RX limits, limit TX to 136-600MHz

Commit:
Spectrum bugfixes and changed height curve

Commit:
Add filter bandwidth setting (narrow/wide FM)

Commit:
cps: Add channel bandwidth setting

Commit:
Fix bug in channel bandwidth setting

Commit:
Added beeps, fixed squelch, more robust audio subsystem

Commit:
Fix inverted ifdef

Commit:
Fix transmitting; speaker was left on, disabling microphone

Commit:
Improve display responsiveness

Trriss committed:
Fixing "RSSI to s_level" in gfx_drawSmeter and gfx_drawSmeterLevel

Trriss committed:
cosmetic - cleaned some spaces

Trriss committed:
RSSI Bar

Commit:
Fix RSSI flickering by persisting last RSSI

Commit:
Colorize RSSI bar

Commit:
General fixes to audio system

Commit:
Fix compile error for other platforms

Commit:
Revert audio fix (It didn't work)

Commit:
Remove unused functions from ST7735S header file

Commit:
Attempting to fix source tree after rebase. Partial implementation of new APIs

Trriss committed:
-Ported spi_stm32 to spi_gd32\n-Created a spiDevice nvm_spi to be used in nvmem_A36Plus\n-Updated nvmem_A36_plus to use the new spi structure and to use gpio_setMode\n-Updated gpio_setMode in gpio_gd32 to it is similar to the stm32 version that can set alternate mode in one go

Trriss committed:
added W25Qx_readData wrapper in nvmem_A36plus and cps_io_native_A36plus

-added W25Qx_erase wrapper in nvmem_A36plus

fixed typo

Trriss committed:
added W25Qx_writeData wrapper & missing return in W25Qx_erase

-added spi_gd32c to meson

Trriss committed:
Try2 updating spiDevice for nvmem & porting to gd32

Trriss committed:
Add commented SPI_BITBANG_DEVICE for bk4819, for future processing

Trriss committed:
Trying to integrated A36Plus with modified MCU: GD32F303 into the regular build, using an extra target: openrtx_a36plusf303 in meson.build, next to the regular A35.

Except for the mcu drivers, and a few additional files (_f303 in the filename), most files are in common with the regular A36.

Trriss committed:
adding flag in pinmap.h for gd32f303cct dev board (pb1 is not accessible on cheap devboard)

linker_script_f303cct6.ld is for cct6 that has only 48k of sram)

Commit:
Fix rebase artifacts to enable compiling again

Commit:
Add missing ifdefs for compliance with the other platforms

Commit:
a36plus: Fix AM/FM switch in macro menu

Commit:
[a36plus] fix bug with changing modulation automatically opening squelch

silseva commented:
To be evaluated what to do, not stale
